streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedStreamsTasks.java

```diff
@@ -325,13 +321,7 @@ boolean maybeResumeSuspendedTask(final TaskId taskId,
                     // we need to catch migration exception internally since this function
                     // is triggered in the rebalance callback
                     log.info("Failed to resume {} {} since it got migrated to another thread already. " +
-                        "Closing it as zombie before triggering a new rebalance.", taskTypeName, task.id());
-                    final RuntimeException fatalException = closeZombieTask(task);
-                    running.remove(taskId);
-
-                    if (fatalException != null) {
-                        throw fatalException;
-                    }
+                             "Will trigger a new rebalance and close all tasks as zombies together.", taskTypeName, task.id());
```

Comment:
Found some other occurrences of closing a zombie and only partially clearing state, updated them all so that all tasks will be closed as zombies together during `onPartitionsLost`